cl_manycore/libraries/bsg_manycore_origin_eva_map.cpp

```diff
@@ -0,0 +1,84 @@
+#include <bsg_manycore_origin_eva_map.h>
+#include <bsg_manycore_eva.h>
+#include <bsg_manycore_printing.h>
+#include <cstring>
```

Comment:
Can you use the ifdef __cplusplus macro to switch between c headers and c++ headers?